internal/resolution/client/osv_offline_client.go

```diff
@@ -0,0 +1,94 @@
+package client
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"sync"
+
+	"deps.dev/util/resolve"
+	"github.com/google/osv-scanner/internal/local"
+	"github.com/google/osv-scanner/internal/resolution/util"
+	"github.com/google/osv-scanner/pkg/models"
+	"github.com/google/osv-scanner/pkg/osv"
+)
+
+type OSVOfflineClient struct {
+	// TODO: OSV-Scanner v2 plans to make vulnerability clients that can be used here.
+	offline     bool
+	localDBPath string
+	mu          sync.Mutex
+}
+
+func NewOSVOfflineClient(downloadDBs bool, localDBPath string) *OSVOfflineClient {
+	return &OSVOfflineClient{
+		offline:     !downloadDBs,
+		localDBPath: localDBPath,
+	}
+}
+
+func (c *OSVOfflineClient) FindVulns(g *resolve.Graph) ([]models.Vulnerabilities, error) {
+	var query osv.BatchedQuery
+	query.Queries = make([]*osv.Query, len(g.Nodes)-1)
+	for i, node := range g.Nodes[1:] {
+		query.Queries[i] = &osv.Query{
+			Package: osv.Package{
+				Name:      node.Version.Name,
+				Ecosystem: string(util.OSVEcosystem[node.Version.System]),
+			},
+			Version: node.Version.Version,
+		}
+	}
+
+	// local.MakeRequest only logs an error if the database cannot be found.
+	// For guided remediation this should be a fatal error, since there's only the one ecosystem.
+	// Set up a reporter to capture error logs and return the logs as an error.
+	r := &errorReporter{}
```

Comment:
I think we should at least try to preserve or replicate the "Loaded xyz local db from path/to/db" call, as knowing that path can be useful especially for offline mode

Reply:
Propagated the log from the first request in `NewOSVOfflineClient`. 
Keeping all of them would be too noisy since the database ends up being loaded after every dependency resolution we make.